src/elements/content-sidebar/versions/VersionsItem.js

```diff
@@ -49,24 +61,22 @@ const VersionsItem = ({ isCurrent, match, permissions, version }: Props) => {
         version_number: versionNumber,
     } = version;
     const isDeleted = action === VERSION_DELETE_ACTION;
-    const className = classNames('bcs-VersionsItem', {
-        'bcs-is-disabled': isDeleted,
-    });
-    const handler = () => {}; // TODO: Add actual event handling logic
-    const versionPath = generatePath(match.path, { ...match.params, versionId });
-    const versionUser = modifiedBy.name || <FormattedMessage {...messages.versionUserUnknown} />;
+    const isDisabled = isDeleted || !permissions.can_preview;
+
+    // Version info helpers
+    const versionSize = sizeUtil(size);
     const versionTimestamp = createdAt && new Date(createdAt).getTime();
+    const versionUser = modifiedBy.name || <FormattedMessage {...messages.versionUserUnknown} />;
+
+    // Version action helper
+    const handleAction = (handler?: (versionId: string) => void) => (): void => handler && handler(versionId);
```

Comment:
Could this be defined outside of `VersionsItem` above like `getActionMessage`?

Reply:
I don't think so. It needs to be a closure because it passes back the `versionId` prop, so it's intended to be per-instance.